accel/opencl/ops_opencl.py

```diff
@@ -223,6 +257,14 @@ def _processing_op(ret, bufs: List[Tuple[str, OpenCLBuffer]]=[], code:str="acc",
 
     replacements = get_replacements(elementwise_prefix, ewtypes)
 
+    (x.image, w.image, ret.image)
+    # fix sampling
+    replacements["INPUT_LOCATION"] = x._image.pos_to_sample_pos("inputLocation")
+    replacements["WEIGHT_LOCATION"] = w._image.pos_to_sample_pos("weightLocation")
+    replacements["OUTPUT_LOCATION"] = ret._image.pos_to_sample_pos("outputLocation", check_bounds=False)
```

Comment:
Why does the output not have to check bounds?

Reply:
Hmm, my impression was that out-of-bounds writes are UB anyway ([ctrl+f undefined](https://man.opencl.org/write_image2d.html)). Whereas for reads - zero padding OOB is expected thanks to `CLK_ADDRESS_CLAMP`.

Reply:
I'm not sure why the output would be different. If you need the bounds check for the input, you are likely overwriting things for the output

Reply:
`check_bounds` may need a better name... The intent is:
* Out-of-bounds `read_imagef` is explicitly allowed in OpenCL.
* Out-of-bounds `write_imagef` is explicitly disallowed (undefined behavior) in OpenCL.
* The replacements on this line rewrite various `int2` locations passed to OpenCL image read / write calls, so that the logical sampling locations map to appropriate locations in the underlying images (after tiling is taken into account).
* Because out-of-bounds `read_imagef` is explicitly allowed in OpenCL, we need to do extra work to make sure that out-of-bounds locations also end up out-of-bounds after remapping reads. This extra work is what `check_bounds=True` does (it detects out-of-bounds queries and always maps them to `(-1, -1)`).
* Because out-of-bounds `write_imagef` is explicitly disallowed in OpenCL, an out-of-bounds `outputLocation` should never occur, and if `outputLocation` was out-of-bounds, remapping it to `(-1, -1)` would be broken anyway. So we skip the extra "map out-of-bounds locations to (-1, -1)" work for `outputLocation`, by setting `check_bounds=False`.

I expect the out-of-bounds read get used in padded conv kernels (`short startX = mad24(mad24(startOutputColumn, strideX, -paddingX), totalNumPackedInputChannels,...`), but nothing should be using out of bounds writes.